### DISCUSSION TITLE
Autocompletion

### DISCUSSION THREAD

**Discussion by korchizhinskiy** [NvChad/NvChad#1667] [Q&A]
How to turn off auto-complete when writing text?
I want to enable auto-completion for Space + any key.
In my last configuration I set autocompletion = false.
Doesn't work here, nvim.schedule error

**Comment by johnallsup** [comment]
I'll echo this question. I want to ask the same thing. I would like to have completion require e.g. shift-enter or shift-tab or ctrl-something. When I am typing, I like  what-you-type-is-what-you-get.

In addition, how do you disable auto-close brackets?

**Comment by johnallsup** [comment]
There's probably a right way to do it, and this is probably not it, but here goes. (Here $CONFIG means e.g. ~/.config/nvim)

Edit $CONFIG/lua/plugins/configs/cmp.lua and change the line
```
    ["<CR>"] = cmp.mapping.confirm {
```
to
```
    ["<C-y>"] = cmp.mapping.confirm {
```
so that you press Ctrl-Y to confirm completion rather than CR.

To disable autopair, edit
lua/plugins/init.lua
and comment out/delete the lines
```
--      {
--        "windwp/nvim-autopairs",
--        opts = {
--          fast_wrap = {},
--          disable_filetype = { "TelescopePrompt", "vim" },
--        },
--        config = function(_, opts)
--          require("nvim-autopairs").setup(opts)
--
--          -- setup cmp for autopairs
--          local cmp_autopairs = require "nvim-autopairs.completion.cmp"
--          require("cmp").event:on("confirm_done", cmp_autopairs.on_confirm_done())
--        end,
--      },
```

Generally, you can use :imap to see insert mode mappings and that nicely tells you which file is responsible for that mapping. So you can find where to disable certain things that way. (That's what I did. Somewhere I read that you shouldn't mess with stuff outside the `lua/custom` folders, so I'd welcome someone explaining the right way to disable autopair and restore normal behaviour to CR.)

**Comment by carlisia** [comment]
This post is a bit old but the question just as valid as of today. In case anyone, like me, lands here first, here's documentation for how to do this, it works for me:

https://github.com/hrsh7th/nvim-cmp/wiki/Advanced-techniques#disabling-completion-in-certain-contexts-such-as-comments
